### ISSUE TITLE
openapi 3.1.0 syntax in a 3.0.0 document

### ISSUE THREAD

**Issue by iiian** [nestjs/swagger#1880]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

**TLDR: `@nestjs/swagger` is meant to build an openapi 3.0.0 document (AFAICT), but _may in fact_ output 3.1.0 syntax in certain situations.**

when trying to model an api property of type

```typescript
{ [key: string]: string|boolean };
```

for instance:

```typescript
export class AppConfigurationDto {
  @ApiProperty()
  version: string;
  @ApiProperty({
    type: 'object',
    additionalProperties: { oneOf: [ {type: 'string'}, {type: 'boolean'}] }
  })
  properties: { [property: string]: string|boolean; };
}
```

The [json schema documentation](https://cswr.github.io/JsonSchema/spec/multiple_types/) seems to propose that this shall be modeled as: `{ type: ['string', 'boolean'] }` for openapi@3.1.0, and as `{ oneOf: [ {type:'string'},{type:'boolean'} ] }` for openapi@3.0.0. However:

When I inspect the generated openapi.json file, at the very top I have 

```yaml
#.openapi.yml
openapi: 3.0.0
```

but later on in the same document, the above `AppConfigurationDto`'s `properties` field is modeled as:

```yaml
AppConfigurationDto:
  type: object
  properties:
    version:
      type: string
    properties:
      type: object
      additionalProperties:
        type:       #
          - string  # <-- openapi@3.1.0 syntax usage
          - boolean #
```

I am using this openapi schema to generate a client. The client-generator is `brew install openapi-generator`. Unfortunately, it also looks like openapi-generator is failing to consume `AppConfigurationDto` even if I manually modify it to use the syntax which is considered valid by https://editor.swagger.io/. But, _swagger editor_ currently rejects the output produced natively by @nestjs/swagger in this certain situation.

### Minimum reproduction code

I don't have time to do this unfortunately :( I understand how that may affect priority

### Expected behavior

I'd hope it's possible to model a union of simple types in swagger json.

### Package version

~5.2.1

### NestJS version

^8.0.0

### Node.js version

14.19.1

### In which operating systems have you tested?

- [ ] macOS
- [X] Windows
- [X] Linux

### Other

Ubuntu 20.04 on WSL 2 x Windows 10

**Comment by iiian** [comment]
It's not super clear what openapi-generator even expects here. I can get around the problem by specifying that the `additionalProperties` can just map to `any`.
